pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -106,8 +120,8 @@ func (p *Pod) Suspend() {
 
 // RunWithPodSetsInfo will inject the node affinity and podSet counts extracting from workload to job and unsuspend it.
 func (p *Pod) RunWithPodSetsInfo(podSetsInfo []jobframework.PodSetInfo) error {
-	if len(podSetsInfo) != 1 {
-		return fmt.Errorf("%w: expecting 1 got %d", jobframework.ErrInvalidPodsetInfo, len(podSetsInfo))
+	if p.groupName() == "" && len(podSetsInfo) != 1 {
+		return fmt.Errorf("%w: expecting 1 pod set got %d", jobframework.ErrInvalidPodsetInfo, len(podSetsInfo))
 	}
 	idx := p.gateIndex()
```

Comment:
So this unblocks every pod because the reconciler is called for each of them?

Reply:
Yes, this will run for every pod in the group.